services/construction_service.go

```diff
@@ -841,6 +841,14 @@ func constructContractCallData(methodSig string, methodArgs []string) ([]byte, e
 			{
 				argData = common.HexToAddress(methodArgs[i])
 			}
+		case v == "uint32":
+			{
+				u64, err := strconv.ParseUint(methodArgs[i], 10, 32)
```

Comment:
This case is already handled on L852. You can simply handle the big.Int's `SetString` failure condition on L855 to do the same thing.

Reply:
The failure does not come from the SetString on big.Int.

The error was raised by `arguments.PackValues(argumentsData)` here: https://github.com/Inphi/optimism-rosetta/pull/42/files#diff-170f821dd6c0e1fadd80eb4252115124a573a20b45df78ec2b65ab873c1b327cR880

The root cause was because we are parsing a uint32 as a big.Int, and `PackValues` throws error because of type mismatch.

Reply:
I see. Thanks for explaining.